src/ResourceManager/Network/Commands.Network/VirtualNetwork/Subnet/Delegation/NewAzureDelegationCommand.cs

```diff
@@ -0,0 +1,47 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.VirtualNetwork.Subnet
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Delegation", SupportsShouldProcess = false, DefaultParameterSetName = "SetByResource"), OutputType(typeof(PSDelegation))]
+    public class NewAzureDelegationCommand : AzureSubnetDelegationBase
+    {
+        [Parameter(
```

Comment:
I don't see the point of `AzureSubnetDelegationBase` Existing given that the properties required by individual cmdlets are added to each cmdlet. Just remove the new base class and directly use `NetworkBaseCmdlet`

Motreover this cmndlet and Add-**Delegation cmdlets are themselves not needed.

Reply:
I added it mostly out of convention from what I saw in other cmdlets, but I see value of keeping it in case there are more properties added to Delegations in the future.
I disagree that the cmdlets are not required, please see my responses to other comments for why,